packages/discovery-react-components/src/components/TooltipHighlight/TooltipHighlight.tsx

```diff
@@ -0,0 +1,176 @@
+  isOpen: boolean;
+}
+
+type Props = {
+  //Parent div element
+  parentDiv: React.MutableRefObject<HTMLDivElement | null>;
+
+  // State of the highlight-tootip
+  tooltipAction: TooltipAction;
+};
+
+const baseTooltipPlaceContent = `${settings.prefix}--tooltip-place-content`;
+const baseTooltipCustomContent = `${settings.prefix}--tooltip-custom-content`;
+const baseTooltipContentHeader = `${settings.prefix}--tooltip-content-header`;
+const baseTooltipBoxColor = `${settings.prefix}--tooltip-box-color`;
+const baseTooltipContentCell = `${settings.prefix}--tooltip-content-cell`;
+const baseTooltipContentCellBuffer = `${settings.prefix}--tooltip-content-cell-buffer`;
+
+export const TooltipHighlight: FC<Props> = ({ parentDiv, tooltipAction }) => {
+  const [tooltipInfo, setTooltipInfo] = useState<TooltipInfo>({
+    rectTooltipArea: new DOMRect(),
+    tooltipContent: <div></div>,
+    isOpen: false
+  });
+
+  useEffect(() => {
+    // Compute parameters to overlay the tooltip over the active item
+    const highlightDivRect = parentDiv.current?.getBoundingClientRect() || new DOMRect();
+    const isOpen = tooltipAction.tooltipEvent !== TooltipEvent.LEAVE;
+    const clickRect = tooltipAction.rectActiveElement || new DOMRect();
+    const tooltipRect = new DOMRect(
+      clickRect.x - highlightDivRect.x,
+      clickRect.y - highlightDivRect.y,
+      clickRect?.width,
+      clickRect?.height
+    );
+    const tooltipUpdate = {
+      rectTooltipArea: tooltipRect,
+      tooltipContent: tooltipAction.tooltipContent || <div></div>,
+      isOpen: !!tooltipAction.tooltipContent && isOpen
+    };
+    setTooltipInfo(tooltipUpdate);
+  }, [tooltipAction, setTooltipInfo]);
+
+  return (
+    // Outter div is required to provide tooltip element with position information
+    // "pointerEvents" = "none" so that underlying elements can react to mouse events
+    <div
+      className={cx(baseTooltipPlaceContent)}
+      style={{
+        top: tooltipInfo.rectTooltipArea.y,
+        left: tooltipInfo.rectTooltipArea.x
+      }}
+    >
+      <Tooltip
+        autoOrientation={true}
+        tabIndex={0}
+        showIcon={false}
+        open={tooltipInfo.isOpen}
+        triggerText={
+          <div
+            style={{
+              // Provide size so that tooltip knows boundry where to draw the tooltip
+              width: tooltipInfo.rectTooltipArea.width,
+              height: tooltipInfo.rectTooltipArea.height,
+              pointerEvents: 'none'
+            }}
+          />
+        }
+        children={<div>{tooltipInfo.tooltipContent}</div>}
+      />
+    </div>
+  );
+};
+
+export function calcToolTipContent(
+  facetInfoMap: FacetInfoMap,
+  facetId: string,
+  enrichValue: string
+) {
+  const tableContent = [];
+  let enrichColor = '';
+  let enrichFacetDisplayname = '';
+  if (facetInfoMap[facetId]) {
+    enrichColor = facetInfoMap[facetId].color;
+    enrichFacetDisplayname = facetInfoMap[facetId].displayName;
```

Comment:
This line is throwing a couple errors for me on Chrome when I view the result preview (not advance doc view) for a document. We probably need to add some logic that can tell the parent component to use tooltips or not

<img width="1748" alt="Screenshot 2023-08-22 at 10 53 14 AM" src="https://github.com/watson-developer-cloud/discovery-components/assets/11284492/2c1caab3-4adf-4bc3-baeb-932390f3a613">